### PR TITLE
Cart api

### PR DESCRIPTION
**Pull request by andrzejewsky**

### CODE REVIEW

File: packages/commercetools/api-client/src/api/createCart/defaultMutation.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  mutation createCart($draft: CartDraft!, $locale: Locale!, $storeKey: KeyReferenceInput) {
+    cart: createCart(draft: $draft, storeKey: $storeKey) {
+      id
+      customerId
+      customerEmail
+      lineItems {
+        id
+        productId
```
Comment (filrak) [inline packages/commercetools/api-client/src/api/createCart/defaultMutation.ts:24]: can we add attributes here as well?

Comment (filrak) [reply packages/commercetools/api-client/src/api/createCart/defaultMutation.ts:24]: use case -> configuring products from cart

File: packages/commercetools/api-client/src/index.ts [hunk 2]
```diff
@@ -17,8 +22,21 @@ const setup = <TCacheShape>(setupConfig?: SetupConfig<TCacheShape>): ApolloClien
   })
   locale = setupConfig.locale
   currency = setupConfig.currency
+  productAttributesIncluded = setupConfig.productAttributesIncluded
```
Comment (filrak) [inline packages/commercetools/api-client/src/index.ts]: 1. It needs to be global, not scoped to cart
2. By default there shouldn't be any limit, field should be optional

File: packages/commercetools/api-client/src/types/Api.ts [hunk 3]
```diff
@@ -1,24 +1,34 @@
-interface CustomQuery {
+import { ApolloQueryResult } from 'apollo-client'
+import { FetchResult } from 'apollo-link'
+import { Cart } from './GraphQL'
+
+export interface CustomQuery {
```
Comment (filrak) [inline packages/commercetools/api-client/src/types/Api.ts:5]: i still think custom queries at this point are bad idea, they are breaking all helper-related functionalities and (possibly) composables. If we want to keep it at least log a warining that using custom query can break the functionality

File: packages/commercetools/api-client/tests/setup.ts [hunk 4]
```diff
@@ -3,4 +3,9 @@ import { setup } from './../src/index'
 jest.mock('./../src/helpers/createCommerceToolsLink')
 jest.mock('apollo-client')
 
-setup({ api: null, locale: 'en', currency: 'USD' })
+setup({
+  api: null,
+  locale: 'en',
+  currency: 'USD',
+  productAttributesIncluded: ['color', 'size']
```
Comment (filrak) [inline packages/commercetools/api-client/tests/setup.ts]: do we see any other potential product config options? if so maybe we should put it under `product.attributesIncluded`

File: packages/commercetools/composables/src/useCart/currentCart.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import { getCart, createCart, getStorage } from '@vue-storefront/commercetools-api'
```
Comment (filrak) [inline packages/commercetools/composables/src/useCart/currentCart.ts:1]: shouldnt getStorage be internal helper?

File: packages/commercetools/composables/src/useCart/actions.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { ProductVariant } from "@vue-storefront/commercetools-api/lib/src/types/GraphQL"
+import { UiCartProduct } from '@vue-storefront/interfaces'
+
```
Comment (filrak) [inline packages/commercetools/composables/src/useCart/actions.ts]: please add that 'action| is a name from ct ideally with link, for people new to this platform that will want to maintain this file naming could be tricky

File: packages/commercetools/composables/src/useCart/process.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import { updateCart } from '@vue-storefront/commercetools-api'
```
Comment (filrak) [inline packages/commercetools/composables/src/useCart/process.ts]: As we agreed, lets move those processors to API Client

File: packages/commercetools/composables/tests/mountComposable.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { shallowMount } from '@vue/test-utils'
```
Comment (filrak) [inline packages/commercetools/composables/tests/mountComposable.ts]: if its not a test maybe `_` at the beginning of file to indicate its a helper?

File: packages/core/interfaces/index.ts [hunk 9]
```diff
@@ -34,15 +34,17 @@ export interface UseCart
   COUPON,
   APPLY_COUPON,
   REMOVE_COUPON,
+  UPDATE_QUANTITY,
 > {
   cart: CART;
   addToCart: ADD_TO_CART;
   removeFromCart: REMOVE_FROM_CART;
   clearCart: CLEAR_CART;
+  updateQuantity: UPDATE_QUANTITY;
```
Comment (filrak) [inline packages/core/interfaces/index.ts]: as discussed - just extend composable interface

**Review by filrak — CHANGES_REQUESTED** [review]

File: packages/commercetools/composables/src/useCart/currentCart.ts (hunk 5 above)
Comment (andrzejewsky) [reply packages/commercetools/composables/src/useCart/currentCart.ts:1]: I need this in composables and api-client as well

File: packages/commercetools/api-client/src/types/Api.ts (hunk 3 above)
Comment (andrzejewsky) [reply packages/commercetools/api-client/src/types/Api.ts:5]: this is the old type that was created for category and products api - you can put custom queries there. Of course, before the final release, we have to give a possibility to pass a custom query for all API calls, but of course, it must be handled in the helpers (warning/error)

File: packages/commercetools/api-client/src/api/createCart/defaultMutation.ts (hunk 1 above)
Comment (andrzejewsky) [reply packages/commercetools/api-client/src/api/createCart/defaultMutation.ts:24]: they are in the variant field -> attributesRaw

**Pull request merged**